modules/dnn/src/layers/mvn_layer.cpp

```diff
@@ -350,32 +353,34 @@ class MVNLayerImpl CV_FINAL : public MVNLayer
             {
                 Mat inpRow = inpMat.row(i);
                 Mat outRow = outMat.row(i);
-                float weight = 1.f;
-                float bias = 0.f;
-                if (fuse_batch_norm)
-                {
-                    weight = i < scale.cols ? ((float*)scale.data)[i] : weight;
-                    bias = i < shift.cols ? ((float*)shift.data)[i] : bias;
-                }
+
                 cv::meanStdDev(inpRow, mean, (normVariance) ? dev : noArray());
                 double alpha = 1;
                 if (normVariance)
                 {
                     alpha = 1 / std::sqrt(eps + dev[0]*dev[0]);
                 }
-                double normalizationScale = 1.0;
-                double normalizationShift = 0.0;
-                if (fuse_batch_norm)
+
+                if (scale.empty() || scale.total() == newRows)
```

Comment:
Need to verify that such check is not weak. In example, for input `1x2x2`, and weights `1x2`.